app/actions/remote/entry/common.ts

```diff
@@ -517,3 +519,50 @@ export async function verifyPushProxy(serverUrl: string) {
         // Do nothing
     }
 }
+
+export async function handleEntryAfterLoadNavigation(
+    serverUrl: string,
+    teamMembers: TeamMembership[],
+    channelMembers: ChannelMember[],
+    currentTeamId: string,
+    currentChannelId: string,
+    initialTeamId: string,
+    initialChannelId: string,
+) {
+    try {
+        const {operator, database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const currentTeamIdAfterLoad = await getCurrentTeamId(database);
+        const currentChannelIdAfterLoad = await getCurrentChannelId(database);
+
+        if (currentTeamIdAfterLoad !== currentTeamId) {
+            // Switched teams while loading
+            if (!teamMembers.find((t) => t.team_id === currentTeamIdAfterLoad && t.delete_at === 0)) {
+                await handleKickFromTeam(serverUrl, currentTeamIdAfterLoad);
+            }
+        } else if (currentTeamIdAfterLoad !== initialTeamId) {
+            await handleKickFromTeam(serverUrl, currentTeamIdAfterLoad);
+        } else if (currentChannelIdAfterLoad !== currentChannelId) {
+            // Switched channels while loading
+            if (!channelMembers.find((m) => m.channel_id === currentChannelIdAfterLoad)) {
+                const tabletDevice = await isTablet();
+                const navComponents = NavigationStore.getNavigationComponents();
+                if (tabletDevice || navComponents.includes(Screens.CHANNEL) || navComponents.includes(Screens.THREAD)) {
+                    await handleKickFromChannel(serverUrl, currentChannelIdAfterLoad);
+                } else {
+                    await setCurrentTeamAndChannelId(operator, initialTeamId, initialChannelId);
+                }
+            }
+        } else if (currentChannelIdAfterLoad !== initialChannelId) {
+            const tabletDevice = await isTablet();
+            const navComponents = NavigationStore.getNavigationComponents();
```

Comment:
same as the other comment

```suggestion
            const navComponents = NavigationStore.getScreensInStack();
```